pbxproj/pbxextensions/ProjectFiles.py

```diff
@@ -360,7 +361,8 @@ def remove_file_by_id(self, file_id, target_name=None):
             return True
 
         # remove the file from any groups if there is no reference from any target
-        for group in filter(lambda x: file_ref.get_id() in x.children, self.objects.get_objects_in_section('PBXGroup')):
+        for group in filter(lambda x: file_ref.get_id() in x.children,
+                            self.objects.get_objects_in_section('PBXGroup', 'PBXVariantGroup')):
```

Comment:
Since `.strings` go into `PBXVariantGroup`, do we need to add a new `remove_file_by_id` method, or perhaps add a `remove_file_by_id(..., section='PBXGroup')` parameter?

I haven't decided which one's the best way, so I'm happy to hear.

Reply:
@kronenthaler You're right, the code in this pull request requires the user (the other developers) to know what they're doing. It's not possible "just" add a Localizable.strings, but the multiple operations needs to be called.

Would you mind sharing the branch so we can move it forward? I don't completely understand this project so I'm depending a lot on trial and error and relying on unit tests to cut my way through. If there's a more designed solution, I'm happy to help.

Reply:
@kronenthaler Hi! Thanks for your help and review so far.

I would like to move forward with the `Localizable.strings` support in this repository and I wanted to know what's your plan? We can go with either of the following direction:

 - Add the bare minimum changes that doesn't break this package and keep the rest in our [openedx/openedx-app-ios](https://github.com/openedx/openedx-app-ios/) repository scripts.
 - Add full support for the `Localizable.strings` and `PBXVariantGroup` so the users like [openedx/openedx-app-ios](https://github.com/openedx/openedx-app-ios/) have a clean interface like `add_file_variant(base="Localizable.strings", variant_name="uk", path="Profile/uk.lproj/Localizable.strings")` and `remove_file_variant(base="Localizable.strings", variant_name="uk", path="Profile/uk.lproj/Localizable.strings")` or `remove_file_variants(base="Localizable.strings")`

I have checked the related work so far like https://github.com/kronenthaler/mod-pbxproj/pull/124/files and https://github.com/kronenthaler/mod-pbxproj/compare/master...variant-groups and I'm looking forward to learn more about them once you think they're ready.

I hate to admit this again, but I'm developing this on a Linux machine and it's my first encounter of pbx project files which doesn't help me to get the most suitable solution.

Reply:
I have just checked the related work so far like https://github.com/kronenthaler/mod-pbxproj/pull/124/files and https://github.com/kronenthaler/mod-pbxproj/compare/master...variant-groups and I'm looking forward to learn more about them once you think they're ready. From what I see you're going with the cleaner approach of providing an API that fully supports the `PBXVariantGroup` which is great.

I hate to admit this again, but I'm developing this on a Linux machine and it's my first encounter of pbx project files which doesn't help me to get the most suitable solution.